include/mbedtls/ssl.h

```diff
@@ -338,11 +338,15 @@
 #define MBEDTLS_SSL_SRV_CIPHERSUITE_ORDER_SERVER  0
 
 #if defined(MBEDTLS_SSL_PROTO_TLS1_3) && defined(MBEDTLS_SSL_SESSION_TICKETS)
-#if defined(MBEDTLS_SHA384_C)
+#if ( !defined(MBEDTLS_USE_PSA_CRYPTO) && \
+        defined(MBEDTLS_MD_C) && defined(MBEDTLS_SHA256_C) ) || \
+    ( defined(MBEDTLS_USE_PSA_CRYPTO) && defined(PSA_WANT_ALG_SHA_256) )
 #define MBEDTLS_SSL_TLS1_3_TICKET_RESUMPTION_KEY_LEN        48
-#elif defined(MBEDTLS_SHA256_C)
+#elif ( !defined(MBEDTLS_USE_PSA_CRYPTO) && \
```

Comment:
I'm OK with this change, but it would be good to consider again if the `legacy_or_psa.h` macro's shouldn't be public. We use copies of those in more and more headers, example programs, etc...

Reply:
Good point. I've started an internal discussion about that. Regardless of the result, I suggest not blocking this PR on that, but fixing in a follow-up PR if we reach an agreement on making those macros public.